### ISSUE TITLE
New Saige update available. 

### ISSUE THREAD

**Issue by johnshorter** [comorment/containers#27]
https://github.com/weizhouUMICH/SAIGE

@ofrei 

This update may resolve the issues we are having with saige. 

```
We have released a new version 1.0.0 (on March 15, 2022). It has substantial computation efficiency improvements for both Step 1 and Step 2 for single-variant and set-based tests. We have created a new program github page https://github.com/saigegit/SAIGE with the documentation provided https://saigegit.github.io/SAIGE-doc/ The program will be maintained by multiple SAIGE developers there. The docker image has been updated. Please feel free to try the version 1.0.0 and report issues if any.

Thanks!
````

**Comment by ofrei** [comment]
@bayramakdeniz I agree it's a good idea to update the container. I suggest the following steps:

1. Update https://github.com/comorment/gwas/blob/main/containers/saige/Dockerfile so that it uses latest tag from https://hub.docker.com/r/wzhou88/saige/tags

2. Using our NREC instance, re-build saige.sif file, and update it here:
https://github.com/comorment/containers/tree/main/singularity
Make sure to also update 
https://github.com/comorment/containers/blob/main/CHANGELOG.md

3. Follow steps described here https://github.com/comorment/containers/blob/main/usecases/gwas_demo.md  to validate that saige 1.0.1 is still compatible with scripts generated by gwas.py. If there are any failures let's discuss and update gwas.py.

4. Please keep an eye on this ticket (https://github.com/saigegit/SAIGE/issues/9 ) . If we'll get an answer that  2nd phase of saige analysis now can use "--threads" argument then we can keep current logic (one job per chromosopme). Otherwise we'll also need to implement step2 in smaller chunks.

**Comment by bayramakdeniz** [comment]
@ofrei @johnshorter  , 1,2 and 3 have been addressed. 

The container is there: https://github.com/comorment/containers/blob/main/singularity/saige.sif

New Saige container (v1.02) works as previous version worked that we used (v 0.44) for the following analysis (https://github.com/comorment/containers/blob/main/usecases/gwas_demo.md)

singularity exec --home $PWD:/home $SIF/python3.sif python gwas.py gwas \
--argsfile /REF/examples/regenie/example_3chr_vcf.argsfile \
--pheno CASE CASE2 --covar PC1 PC2 BATCH --analysis saige figures --out run3_saige

Note that, the current analysis assumes plink format for step1 and vcf format for step 2 as input.  I am examining how to add plink format option for both steps.

**Comment by ofrei** [comment]
@bayramakdeniz thanks! As you pointed out, SAIGE is being actively developed at the moment. I think we should postpone switching to saige 1.0 until it became more stable. Besides, we got a confirmation that step2 is still single-threaded, therefore gwas.py must split step2 SAIGE jobs of each chromosome into many chunks. I'll implement this in gwas.py.

The changes that Bayram implemented are now moved to a separate branch: https://github.com/comorment/containers/tree/bayram_march_2022

I rolled back (i.e. removed) those changes from the main branch, because there were some technical issues with few commits. I'm particularly concerned about https://github.com/comorment/containers/commit/c38f807da7e452b478d79fe0895a6feb0471a43b which appears to touch 100+ files - this isn't supposed to happen.

I'll let you know when an updated gwas.py with support for multiple chunks per chromosome is available. Also, let's see how it goes with SAIGE development, and find a good time to pull the new version - now it's too early to do this.
